sdk/src/Services/DynamoDBv2/Custom/DataModel/_async/Context.Async.cs

```diff
@@ -339,70 +339,94 @@ public Task ExecuteTransactWriteAsync(TransactWrite[] transactionParts, Cancella
 
         #region Scan async
 
-        /// <summary>
-        ///  Configures an async Scan operation against DynamoDB, finding items
-        /// that match the specified conditions.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="conditions">
-        /// Conditions that the results should meet.
-        /// </param>
-        /// <param name="operationConfig">Config object which can be used to override that table used.</param>
-        /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
+        /// <inheritdoc/>
+        public AsyncSearch<T> ScanAsync<T>(IEnumerable<ScanCondition> conditions)
+        {
+            var scan = ConvertScan<T>(conditions, null);
+            return FromSearchAsync<T>(scan);
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("Use the ScanAsync overload that takes ScanConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to ScanAsync.")]
         public AsyncSearch<T> ScanAsync<T>(IEnumerable<ScanCondition> conditions, DynamoDBOperationConfig operationConfig = null)
         {
             var scan = ConvertScan<T>(conditions, operationConfig);
             return FromSearchAsync<T>(scan);
         }
 
-        /// <summary>
-        ///  Configures an async Scan operation against DynamoDB, finding items
-        /// that match the specified conditions.
-        /// </summary>
-        /// <typeparam name="T">Type of object.</typeparam>
-        /// <param name="scanConfig">Scan request object.</param>
-        /// <param name="operationConfig">Config object which can be used to override the table used.</param>
-        /// <returns>AsyncSearch which can be used to retrieve DynamoDB data.</returns>
-        public AsyncSearch<T> FromScanAsync<T>(ScanOperationConfig scanConfig, DynamoDBOperationConfig operationConfig = null)
+        /// <inheritdoc/>
+        public AsyncSearch<T> ScanAsync<T>(IEnumerable<ScanCondition> conditions, ScanConfig scanConfig)
+        {
+            var scan = ConvertScan<T>(conditions, scanConfig?.ToDynamoDBOperationConfig());
+            return FromSearchAsync<T>(scan);
+        }
+
+        /// <inheritdoc/>
+        public AsyncSearch<T> FromScanAsync<T>(ScanOperationConfig scanConfig)
         {
             if (scanConfig == null) throw new ArgumentNullException("scanConfig");
 
-            var search = ConvertFromScan<T>(scanConfig, operationConfig);
+            var search = ConvertFromScan<T>(scanConfig, null);
+            return FromSearchAsync<T>(search);
+        }
+
+        /// <inheritdoc/>
+        [Obsolete("Use the FromScanAsync overload that takes ScanConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to FromScanAsync.")]
+        public AsyncSearch<T> FromScanAsync<T>(ScanOperationConfig scanOperationConfig, DynamoDBOperationConfig operationConfig = null)
```

Comment:
Renaming the argument from `scanConfig` -> `scanOperationConfig` would be a breaking change

Reply:
Reset in https://github.com/aws/aws-sdk-net/pull/3383/commits/987fe702d9428657cff4228bf91457a1ae9dcd74, and names are now less ambiguous after the adjustments due to this thread: https://github.com/aws/aws-sdk-net/pull/3383#discussion_r1673120405